core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -198,6 +198,13 @@ object ExternalStoreUtils extends Logging {
     }
   }
 
+  def isSmartConnectorMode(sparkContext: SparkContext): Boolean = {
```

Comment:
Can we create a lazy val for this? The decision that app is smart connector or not can be made once.